WPMedia/fusion-news-theme-blocks
# Tmedia 437/social sign in

## Update (Wed Sept 29)
There is an issue in the Identity SDK that needs to be addressed.  If a site was logged in with Google and then the user signs out of the site, they are still logged in with Google....so when the user comes to a login page, `Identity.initGoogleLogin` will immediately call the `onSuccess` handler provided. 

This is what was causing the endless renders that I talked about....but it wasn't re-renders.  What was happening is that even though I was logged off the _site_, I wasn't logged off _Google_....so when I came to the login page, the `onSuccess` was immediately called and redirected to the /account page, the account page see's that I am not logged onto the _site_ and then takes me back to the login page....and the cycle continues.  

The way I am dealing with this until a fix for Identity is pushed is to just use a ref to reject the initial onSucess call:  See line 44 in blocks/identity-block/components/SocialSignOn/index.jsx:
`
        onSuccess: () => {
          if (!isGoogleReset.current) {
            isGoogleReset.current = true;
          } else {
            window.location = redirectURL;
          }
        },
        onFailure: () => { onError(); },
      });
`

Also, @mattkim93 has added the Facebook credentials to the core components org, so the Facebook social button is now showing up, but you will still need to be under https to actually use it.....at least it allowed me to tweak the styles a bit to get everything properly sized and positioned.

## Update (Fri Sept 24)
Appreciate the generous review and guidance on this pr.  Couple things that I did that I assume was against the ask in the ticket:

1. I added the social sign-on component into the login and sign in features.  This has been removed.
2. I assumed the captcha was needed to make social sign-on work.  This was not the case, it was due to lack of experience with the Identity API on my part.

I believe I have addressed all the concerns in the PR.  One change to the testing, when creating a test page, you will now have to add the Social Sign On feature directly onto the page (ideally below the login or sign up feature...or alone) as the social sign on component has been removed from the sign on and login feature as advised by Matt R.

Also, In regards to testing the Facebook locally.  It's not possible as a https connection is needed.  There is no solution in place.  It is the reason I brought it up in the PB Knowledge Transfer meeting this week.  From listening to Jason and Luke describe it, it's doable, but time needs to be allocated to it.  Also mentioned this to Robin and there will be some momentum to get this on their radar.

## Description
This adds the social media (Google and Facebook) sign on feature and component and integrates this into the login and signup feature for the Identity block.

## Jira Ticket
https://arcpublishing.atlassian.net/browse/TMEDIA-437

## Acceptance Criteria
Google Social Sign On button is visible when Google is selected in Subs
Facebook Social Sign On button is visible when Facebook is selected in Subs

Note: Facebook button is ONLY visible behind https .

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-437/social-sign-in `
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block,@wpmedia/shared-styles`
3. Create a signup page using the Identity Block's sign-in feature.  Verify that at least the Google button renders.
4. Create a login page using the Identity Block's log-in feature.  Verify that at least the Google button renders.
5. Create a social sign-on page using the Identity Block's sSocial Sign On feature.  Verify that at least the Google button renders.


## Dependencies or Side Effects
_Examples of dependencies or side effects are:_ Visual testing of Facebook sucks (it doesn't show) because of the https requirement.....would be interesting to set up local environment under https if its just a simple matter of generating certs...


## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ x] Confirmed all the test steps a reviewer will follow above are working.
- [x ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [x ] Ran `npm run test`, made sure all tests are passing
  - [x ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).    Due to the need of the captcha on the signup and login pages, the tests are complaining about the tests not being in act wrapper...even though they are.  Might be a sign we need to better decouple the useEffect logic on those features.


## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
